// middleware/routes.js
/**
 * Created by mark on 11/12/2015.
 */

module.exports = function (app, fs) {
    app.post('/api/getProducts',function (req, res) {
        var products;
        if(req.body.type == 's') {
            products = [
                {
                    title: "Huawei P6S", description: "", status: ' ', statusText: ' ', img: "/img/main-products/p6s.png", slug: "p6s"
                },
                {
                    title: "Honor 3", description: "", status: ' ', statusText: ' ', img: "/img/main-products/honor-3.png", slug: "honor-3"
                },
                {
                    title: "Huawei G6", description: "", status: ' ', statusText: ' ', img: "/img/main-products/g6-3g.png", slug: "g6-3g"
                },
                {
                    title: "Huawei Y220", description: "", status: ' ', statusText: ' ', img: "/img/main-products/y220.png", slug: "y220"
                },
                {
                    title: "Honor 4C Pro", description: "", status: ' ', statusText: ' ', img: "/img/main-products/4c-pro.png", slug: "4c-pro"
                },
                {
                    title: "Huawei G730", description: "", status: ' ', statusText: ' ', img: "/img/main-products/g730.png", slug: "g730"
                },
                {
                    title: "Huawei P9 Lite", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/p9-lite.png", slug: "p9-lite"
                },
                {
                    title: "Huawei P9", description: "", status: ' ', statusText: ' ', img: "/img/main-products/p9.png", slug: "huawei-p9"
                },
                {
                    title: "Honor 5X", description: "", status: ' ', statusText: ' ', img: "/img/main-products/honor-5x.png", slug: "honor-5x"
                },
                {
                    title: "Huawei Y5 II", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/y5ii.png", slug: "y5ii"
                },
                {
                    title: "Huawei Y3 II", description: "", status: ' ', statusText: ' ', img: "/img/main-products/y3ii.png", slug: "huawei-y3ii"
                },
                {
                    title: "Honor 5C", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/honor-5c.png", slug: "honor-5c"
                },
                {
                    title: "Huawei GR3", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/honor-5c.png", slug: "gr3"
                },
                // {
                //     title: "Honor 8", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/"
                //     //,slug: "honor-8"
                // },
                {
                    title: "Huawei Y6 II", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/y6ii.png", slug: "y6ii"
                }
            ];
            res.send(products).end();
        }
        else if(req.body.type == 't') {
            products = [
                {
                    title: "Huawei MediaPad X1 7.0", description: "", img: "/img/main-products/mediapad-x1-7-0.png", slug: "mediapad-x1-7"
                },
                {
                    title: "Huawei MediaPad T1 7.0", description: "", img: "/img/main-products/mediapad-t1-7-0.png", slug: "mediapad-t1-7-0"
                },
                {
                    title: "Huawei MediaPad T1 8.0", description: "", img: "/img/main-products/mediapad-t1-8-0.png", slug: "mediapad-t1-8-0"
                }
            ];
            res.send(products).end();
        }
    });
    app.post('/api/getProduct',function (req, res) {
        var path = './data/products/'+ req.body.lang + '-' + req.body.slug + '.json';
        fs.readFile(path, 'utf8', function (err, data) {
            if (err) {
                console.error(err);
                res.send({error : 'No such product.'});
            }
            else {
                var obj = JSON.parse(data);
                res.send(obj);
            }

        });
    });
    app.post('/api/getArticles',function (req, res) {
        var articles;
        switch(req.body.lang) {
            case 'en':
                articles = [
                    {
                        title: "Why It's Impossible not to Love Huawei : 11 Facts", text: "It's sure not easy to choose a smartphone or a gadget these days. Options are vast, criteria different. So it's not easy to cherry-pick and as a result you might end up with a device you bought simply because your friends have it. In order to understand the quality criteria one has to be familiar with industry leaders. Chinese company Huawei is one of the biggest smartphone manufacturers nowadays. A few years back this would have been hard to believe but today it's a stated fact. Last year the company introduced its new concept: «Make it Possible», and Huawei's new motto is «Dreams inspire creativity». Below are 11 facts about Huawei out of many that will change you perception of this Chinese company.", date: '11.08.2016', img: '1.png', slug: '11_facts_about_huawei'
                    }
                ];
                break;
            case 'ru':
                articles = [
                    {
                        title: "11 фактов которые вы незнали о Huawei", text: "Выбор смартфонов и другого оборудования на сегодняшний день является нелегкой задачей. Учитывая широкий ассортимент и различные критерии, можно легко запутаться, и при выборе руководствоваться не качеством, а лишь тем, что выбирают наши знакомые. Более того, необходимо знать лидеров данной отрасли для того, чтобы опеределить критерии оценки качества. Китайская компания Huawei является одним из крупнейших производителей смартфонов. Несколько лет назад в это было бы трудно поверить, но на сегодняшний день это является признанным фактом.", date: '11.08.2016', img: '1.png', slug: '11_facts_about_huawei'
                    }
                ];
                break;
            case 'hy':
            default:
                articles = [
                    {
                        title: "11 փաստ, որոնք իմանալով հնարավոր չէ չսիրել Huawei-ը", text: "Սմարթֆոններ եւ այլ տեխնիկա ընտրելն այսօր հեշտ գործ չէ: Ընտրությունը մեծ է,  չափանիշները` տարբեր, ինչի հետեւանքով կարելի է խճճվել ու որակի փոխարեն ընտրել ընդամենը այն, ինչ ընտրում են մեր ծանոթները: Իսկ որակի չափանիշները հասկանալու համար պետք է ծանոթ լինել ոլորտի առաջատարներին:", date: '11.08.2016', img: '1.png', slug: '11_facts_about_huawei'
                    }
                ];
                break;
        }

        res.send(articles).end();
    });
    app.post('/api/getArticle',function (req, res) {
        var path = './data/blog/'+ req.body.lang + '-' + req.body.slug;
        fs.readFile(path + '.json', 'utf8', function (err, data) {
            if (err) {
                console.error(err);
                res.send({error : 'No such article.'});
            }
            else {
                var obj = JSON.parse(data);

                fs.readFile(path + '.html', 'utf8', function (err, data1) {
                    if (err) {
                        console.error(err);
                        res.send({error : 'No such article.'});
                    }
                    else {
                        obj.inner = data1;
                        res.send(obj);
                    }

                });
            }

        });
    });
    //app.post('*', function (req, res) {
    //    res.sendFile('index.html', { root: '././files/' });
    //});
    //app.get('*', function (req, res) {
    //    res.sendFile('index.html', { root: '././files/' });
    //});
    app.post('/sitemap.xml', function (req, res) {exports.generateSitemap(req, res)});

    app.get('/sitemap.xml', function (req, res) {exports.generateSitemap(req, res)});

    app.post('/robots.txt', function (req, res) {exports.generateRobots(req, res)});

    app.get('/robots.txt', function (req, res) {exports.generateRobots(req, res)});

    app.get('*', function (req, res) {
        var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
        if (userAgent != 'Googlebot' && userAgent != 'bingbot' && userAgent != 'Slurp' && userAgent != 'FacebookBot') {
            res.sendFile('index.html', { root: '././files/' });
        }
        else {
            exports.botHack(req, res);
        }
    });

    app.post('*', function (req, res) {
        var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
        if (userAgent != 'Googlebot' && userAgent != 'bingbot' && userAgent != 'Slurp' && userAgent.indexOf('Facebook') == -1 && userAgent.indexOf('facebookexternalhit/1.1') == -1 && userAgent != 'Facebot') {
            res.sendFile('index.html', { root: '././files/' });
        }
        else {
            exports.botHack(req, res);
        }
    });
};


exports.botHack = function (req, res) {
    var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
    var renderObject = {
        test : "test-dev"
    };
    try {
        if (req.url == '') {
            console.log("bot / main");
            req.route.path = 'main';
        }
        else if (req.url == '/') {
            console.log("bot / main");
            req.route.path = 'main';
        }
        else if (req.url.indexOf('blog') > -1) {
            console.log("bot / blog");
            req.route.path = 'blog';
        }
        else if (req.url.indexOf('product') > -1) {
            console.log("bot / product");
            req.route.path = 'product';
        }
        else if (req.url.indexOf('enterprises') > -1) {
            console.log("bot / enterprises");
            req.route.path = 'enterprices';
        }
        else if (req.url.indexOf('aboutus') > -1) {
            console.log("bot / aboutus");
            req.route.path = 'aboutus';
        }
        else if (req.url.indexOf('termsofuse') > -1) {
        }
        else if (req.url.indexOf('privacy') > -1) {
        }
        else if (req.url.indexOf('signup') > -1) {
        }
        else if (req.url.indexOf('genre') > -1) {
        }
        else if (req.url.indexOf('article') > -1) {
        }
        else {
            req.route.path = 'main';
            console.log("bot / other");
        }
        var routeName = req.route.path.indexOf(':') === -1 ? req.route.path : req.route.path.substring(0, req.route.path.indexOf(":") - 1);
        console.log(routeName);
        exports.renderer(req, res, renderObject, routeName);
    }
    catch (e) {
        console.log({'level': 'ERROR', 'message': e});
    }
}

exports.generateSitemap = function(req, res) {
    var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
    var renderObject = {
        locales : ['hy','en','ru'],
        products : [
            {title:"Huawei NEXUS 6P",
                description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"}]
    };
    try {
        exports.renderer(req, res, renderObject, 'sitemap');
    }
    catch (e) {
        console.log({'level': 'ERROR', 'message': e});
    }
}

exports.renderer = function (req, res, rendererObject, from) {
    var hostname = req.headers.host.indexOf(':') === -1 ? req.headers.host : req.headers.host.substring(0, req.headers.host.indexOf(':'));
    try {
        var lang;
        if(req.url.slice(3,4)=="/") lang = req.url.slice(1,3); else lang='hy';
        var builderObject = {}, path, platformData = {}, lang_links=[];
        var metaDesc = {
            'en' : 'Huawei is the world’s largest manufacturer of telecommunication equipment. Visit our site and get all essential information about Huawei smartphones, tablets and hardware. Be the first to know about our new arrivals and hot sales. ',
            'ru' : 'Huawei крупнейший в мире производитель телекоммуникационного оборудования. Получите всю необходимую информацию о Смартфонах, планшетах и других устройствах Huawei. Узнайте первыми о новинках и супер скидках.',
            'hy' : 'Huawei  հեռահաղորդակցության սարքավորումների աշխարհի խոշորագույն արտադրող ընկերությունն է: Այցելե՜ք մեր կայք և ստացեք ամբողջական տեղեկատվություն Huawei հեռախոսների, պլանշետների և այլ սարքավորումների մասին: Առաջինը իմացեք մեծ զեղչերի և նոր արտադրանքի մասին:'
        }
        switch (from) {
            case 'robots':
                builderObject.seo = {
                    hostname : hostname
                }
                path = 'robots';
                break;
            case 'sitemap':
                builderObject.seo = {
                   hostname : hostname,
                   locales : rendererObject.locales,
                   products :  rendererObject.products
                }
                path = 'sitemap';
                break;
            case 'main':
                var descr = {
                    'en' : 'Huawei is the world’s largest manufacturer of telecommunication equipment. Visit our site and get all essential information about Huawei smartphones, tablets and hardware. Be the first to know about our new arrivals and hot sales.',
                    'ru' : 'Huawei крупнейший в мире производитель телекоммуникационного оборудования. Получите всю необходимую информацию о Смартфонах, планшетах и других устройствах Huawei. Узнайте первыми о новинках и супер скидках. ',
                    'hy' : 'Huawei  հեռահաղորդակցության սարքավորումների աշխարհի խոշորագույն արտադրող ընկերությունն է: Այցելե՜ք մեր կայք և ստացեք ամբողջական տեղեկատվություն Huawei հեռախոսների, պլանշետների և այլ սարքավորումների մասին: Առաջինը իմացեք մեծ զեղչերի և նոր արտադրանքի մասին:'
                };
                var keywords = {
                    "en" : "huawei,huawei phones,huawei mobile,huawei honor,huawei tablet,smart phones,huawei smartphone,telephone huawei,huawei telefon,huawei 8,huawei mobile phones,huawei technologies,huawei device,huawei mate 2,huawei shop,best huawei phone,huawei android,huawei latest phone,huawei new phone,huawei review,huawei company,huawei website",
                    "ru" : "Хуавей,Хуавей телефоны,Хуавей мобильники,Хуавей хонор,хуавей планшеты,планшеты,Хуавей смартфоны,смартфоны,телефоны хуавей,телефоны хуавей,хуавей 8,мобильные телефоны,хуавей технология,хуавей продукция,huawei mate 2,хуавей магазин,лучшая цена хуавей,андроид хуавей,последние модели,новинки,отзывы,компания,веб сайт,хуавей,Хуавей Армения,Хуавей Ереван",
                    "hy" : "Խուավեյ,Հուավեյ,huawei,huawei Հեռախոս,huawei Մոբայլ,huawei honor,huawei տաբլետ, սմարտֆոններ,huawei սմարտֆոններ, հեռախոսներ huawei,huawei 8,huawei մոբայլ հեռախոսներ,huawei տեխնոլոգիա, huawei սարքեր,huawei mate 2, huawei խանութ,Լավագույն huawei հեռախոսներ,huawei android,huawei վերջին հեռախոսներ,huawei նոր հեռախոսներ,huawei review,huawei ընկերություն, huawei կայք,Գնել huawei,Առնել huawei"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'smartphones':
                var keywords = {
                    "en" : "huawei phones,huawei mobile,huawei honor,huawei,smart phones,huawei smartphone,huawei ascend,telephone huawei,huawei telefon,huawei 8,huawei mobile phones,huawei phone price,huawei mobile price,huawei device,huawei mate 2,mediapad,huawei e5220,huawei honor 5c,honor huawei,huawei shop,huawei phones price list,best huawei phone,huawei android,huawei price,huawei latest phone,huawei new phone,huawei p8 lite,nexus 6p,huawei mate 8,huawei mate s,huawei mate 7,huawei ascend p7,huawei ascend mate 7,huawei honor 6,huawei g elite,telephone huawei,Huawei Armenia,Huawei Yerevan",
                    "ru" : "Хуавей телефоны,Хуавей хонор,Хуавей,смартфоны,хуавей смартфоны,huawei ascend,телефоны хуавей,Хуавей телефоны,хуавей 8,мобильные телефоны,хуавей цена,хуваей смартфоны цена,хуавей продукция,huawei mate 2,mediapad,huawei e5220,huawei honor 5c,honor huawei,huawei shop,huawei phones price list,лучшая цена хуавей,андроид хуавей,хуавей цена,последние модели,новинки,huawei p8 lite,nexus 6p,huawei mate 8,huawei mate s,huawei mate 7,huawei ascend p7,huawei ascend mate 7,huawei honor 6,huawei g elite,Хуавей Армения,Хуавей Ереван",
                    "hy" : "huawei Հեռախոս,huawei Մոբայլ,huawei honor,huawei,սմարտֆոններ,huawei սմարտֆոններ,huawei ascend,հեռախոսներ huawei,huawei 8,huawei մոբայլ հեռախոսներ,huawei հեռախոսի գներ,huawei մոբայլ հեռախոսներ,huawei սարքեր,huawei mate 2,mediapad,huawei e5220,huawei honor 5c,honor huawei,huawei խանութ,huawei հեռախոսի գներ,Լավագույն huawei հեռախոսներ,huawei android,huawei գներ,huawei վերջին հեռախոսներ,huawei նոր հեռախոսներ,huawei p8 lite,nexus 6p,huawei mate 8,huawei mate s,huawei mate 7,huawei ascend p7,huawei ascend mate 7,huawei honor 6,huawei g elite,telephone huawei,Huawei Հայաստան,Huawei Երևան,huawei կարծիքներ,huawei կարծիք,huawei արտադրանք,Գնել huawei,Առնել huawei"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'tablets':
                var keywords = {
                    "en" : "huawei,huawei tablet,huawei mediapad,huawei technologies,huawei device,huawei tablet price,mediapad,huawei tablet price list,huawei shop,Huawei Armenia,Huawei Yerevan,huawei review",
                    "ru" : "Хуавей,планшеты,huawei mediapad,хуавей технология,хуавей продукция,хуавей планшеты цена,mediapad,отзывы,хуавей магазин,Huawei Armenia,Huawei Yerevan,huawei review,Хуавей Армения,Хуавей Ереван",
                    "hy" : "huawei,huawei տաբլետ,huawei mediapad,huawei տեխնոլոգիա,huawei device,huawei պլանշետ գին,mediapad,huawei տաբլետ գին,huawei խանութ,Huawei Հայաստան,Huawei Երևան,huawei կարծիք,Խուավեյ,Հուավեյ,Huawei պլանշետներ,Huawei պլանշետների գներ,huawei խանութ,Գնել huawei,Առնել huawei"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'blog':
                var keywords = {
                    "en" : "huawei phones,huawei mobile,huawei tablet,smart phones,Huawei Armenia,Huawei Yerevan",
                    "ru" : "Хуавей телефоны,хуавей магазин,хуавей планшеты,smart phones,Huawei Армения,Huawei Ереван",
                    "hy" : "huawei Հեռախոս,huawei Մոբայլ,Huawei պլանշետներ,huawei սմարտֆոններ,Huawei Հայաստան,Huawei Երևան,Խուավեյ,Հուավեյ,huawei վերջին հեռախոսներ,huawei նոր հեռախոսներ,Huawei նորություններ,Huawei պլանշետներ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Huawei Blog",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'enterprises':
                var keywords = {
                    "en" : "huawei,huawei phones,huawei device,huawei technologies co ltd,Huawei become a partner,Huawei partner,Huawei Mobile Phones partiler,huawei shop,Huawei Armenia,Huawei Yerevan,huawei review",
                    "ru" : "Хуавей,Хуавей телефоны,хуавей устройства,хуавей технологии,Huawei для партнеров,Huawei партнерство,Huawei мобильные устройства,хуавей магазин,Huawei Армения,Huawei Ереван,huawei отзывы",
                    "hy" : "huawei,huawei Հեռախոս,huawei սարքեր,huawei արտադրանք,huawei տաբլետ,Huawei դարձիր գործընկեր,Huawei գործընկեր,huawei սմարտֆոններ,huawei խանութ,Huawei Հայաստան,Huawei Երևան,Խուավեյ,Հուավեյ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Enterprises",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'aboutus':
                var keywords = {
                    "en" : "huawei,contacts,contact us,huawei mobile,huawei tablet,huawei shop,huawei technologies co ltd,Huawei Armenia,Huawei Yerevan,huawei company,about Huawei,about us",
                    "ru" : "Хуавей,контакты,свяжитесь с нами,Хуавей телефоны,хуавей планшеты,хуавей магазин,хуавей устройства,Huawei Армения,Huawei Ереван,компания Huawei,про компанию Huawei,о нас",
                    "hy" : "մեր մասին,կապ,Huawei ընկերություն,Huawei ներկայացուցիչ,huawei խանութ,huawei խանութ,Huawei մասին,Huawei Հայաստան,Huawei Երևան,Խուավեյ,Հուավեյ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "About",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'ourpartners':
                var keywords = {
                    "en" : "Huawei become a partner,Huawei partner,Huawei Mobile Phones partiler,huawei shop,Huawei Armenia,Huawei Yerevan,huawei review",
                    "ru" : "Huawei стань партнером,Huawei партнеры,Huawei Mobile Phones partiler,huawei shop,Huawei Армения,Huawei Ереван,huawei отзывы,Huawei стань представителем",
                    "hy" : "Huawei դարձիր գործընկեր,Huawei գործընկեր,Huawei Mobile Phones partiler,huawei խանութ,huawei կարծիք,Huawei Հայաստան,Huawei Երևան,Խուավեյ,Հուավեյ,գործընկերներ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Partners",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'contacts':
                var keywords = {
                    "en" : "huawei,contacts,contact us,huawei mobile,huawei tablet,huawei shop,huawei technologies co ltd,Huawei Armenia,Huawei Yerevan,huawei company,Huawei address,Huawei ardess,Huawei Map,Huawei number,Huawei office",
                    "ru" : "huawei,контакты,свяжитесь с нами,Хуавей телефоны,хуавей планшеты,хуавей магазин,хуавей устройства,Huawei Армения,Huawei Ереван,компания Huawei,Huawei адресс,Huawei адресс магазина,Huawei Карта,Huawei номер телефона,Huawei офис",
                    "hy" : "հասցե,կապ,հեռախոս,huawei ընկերություն,Huawei հասցե,huawei խանութ,Huawei խանութ,Huawei հեռախոսահամար,Huawei գրասենյակ,Huawei Հայաստան,Huawei Երևան,Խուավեյ,Հուավեյ,huawei Հեռախոս"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Contacts",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'product':
                console.log('aa')
                break;
        }
        res.render('' + path + '.html', {data: builderObject}, function (error, html) {
            if (!error) {
                res.end(html);
                //res.end();
                return;
            }
            else {
                console.error("renderer error : " + error);
            }
            return;
        });
    }
    catch (e) {
        console.log(e);
    }
}
