// middleware/routes.js
/**
 * Created by mark on 11/12/2015.
 */

module.exports = function (app, fs) {
    app.post('/api/getProduct',function (req, res) {
        var path = './data/products/'+ req.body.lang + '-' + req.body.slug + '.json';
        fs.readFile(path, 'utf8', function (err, data) {
            if (err) {
                console.error(err);
                res.send({error : 'No such product.'});
            }
            else {
                var obj = JSON.parse(data);
                res.send(obj);
            }

        });
        //try {
        //    var query = productsSchema.find({
        //        slug : req.body.slug,
        //        lang : req.body.lang
        //    }).exec(function (err, docs) {
        //        if (!err) {
        //            if(docs.length == 1) {
        //                res.send({error : false, data : docs[0], message : "success"}).end();
        //            }
        //            else {
        //                res.send({error : true, message : "product-error", code : "pe1"}).end();
        //            }
        //        }
        //        else {
        //            console.log('error');
        //        }
        //    })
        //}
        //catch (err) {
        //    console.error(err);
        //}
    });
    //app.post('*', function (req, res) {
    //    res.sendFile('index.html', { root: '././files/' });
    //});
    //app.get('*', function (req, res) {
    //    res.sendFile('index.html', { root: '././files/' });
    //});
    app.post('/sitemap.xml', function (req, res) {exports.generateSitemap(req, res)});

    app.get('/sitemap.xml', function (req, res) {exports.generateSitemap(req, res)});

    app.post('/robots.txt', function (req, res) {exports.generateRobots(req, res)});

    app.get('/robots.txt', function (req, res) {exports.generateRobots(req, res)});

    app.get('*', function (req, res) {
        var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
        console.log('agent - ' + userAgent);
        if (userAgent != 'Googlebot' && userAgent != 'bingbot' && userAgent != 'Slurp' && userAgent != 'FacebookBot') {
            res.sendFile('index.html', { root: '././files/' });
        }
        else {
            console.log('bot - ' + userAgent);
            exports.botHack(req, res);
            //exports.botHack(req, res); req._parsedOriginalUrl.pathname == '/' &&
        }
    });

    app.post('*', function (req, res) {
        var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
        console.log('agent - ' + userAgent);
        if (userAgent != 'Googlebot' && userAgent != 'bingbot' && userAgent != 'Slurp' && userAgent.indexOf('Facebook') == -1 && userAgent.indexOf('facebookexternalhit/1.1') == -1 && userAgent != 'Facebot') {
            res.sendFile('index.html', { root: '././files/' });
        }
        else {
            console.log('bot - ' + userAgent);
            exports.botHack(req, res);
            //exports.botHack(req, res); req._parsedOriginalUrl.pathname == '/' &&
        }
    });
};


exports.botHack = function (req, res) {
    var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
    var renderObject = {
        test : "test-dev"
    };
    try {
        if (req.url == '') {
            console.log("bot / main");
            req.route.path = 'main';
        }
        else if (req.url == '/') {
            console.log("bot / main");
            req.route.path = 'main';
        }
        else if (req.url.indexOf('blog') > -1) {
            console.log("bot / blog");
            req.route.path = 'blog';
        }
        else if (req.url.indexOf('product') > -1) {
            console.log("bot / product");
            req.route.path = 'product';
        }
        else if (req.url.indexOf('enterprises') > -1) {
            console.log("bot / enterprises");
            req.route.path = 'enterprices';
        }
        else if (req.url.indexOf('aboutus') > -1) {
            console.log("bot / aboutus");
            req.route.path = 'aboutus';
        }
        else if (req.url.indexOf('termsofuse') > -1) {
        }
        else if (req.url.indexOf('privacy') > -1) {
        }
        else if (req.url.indexOf('signup') > -1) {
        }
        else if (req.url.indexOf('genre') > -1) {
        }
        else if (req.url.indexOf('article') > -1) {
        }
        else {
            req.route.path = 'main';
            console.log("bot / other");
        }
        var routeName = req.route.path.indexOf(':') === -1 ? req.route.path : req.route.path.substring(0, req.route.path.indexOf(":") - 1);
        console.log(routeName);
        exports.renderer(req, res, renderObject, routeName);
    }
    catch (e) {
        console.log({'level': 'ERROR', 'message': e});
    }
}

exports.generateSitemap = function(req, res) {
    var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
    var renderObject = {
        locales : ['hy','en','ru'],
        products : [
            {title:"Huawei NEXUS 6P",
                description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"}]
    };
    try {
        exports.renderer(req, res, renderObject, 'sitemap');
    }
    catch (e) {
        console.log({'level': 'ERROR', 'message': e});
    }
}

exports.renderer = function (req, res, rendererObject, from) {
    var hostname = req.headers.host.indexOf(':') === -1 ? req.headers.host : req.headers.host.substring(0, req.headers.host.indexOf(':'));
    try {
        var lang;
        if(req.url.slice(3,4)=="/") lang = req.url.slice(1,3); else lang='hy';
        var builderObject = {}, path, platformData = {}, lang_links=[];
        switch (from) {
            case 'robots':
                builderObject.seo = {
                    hostname : hostname
                }
                path = 'robots';
                break;
            case 'sitemap':
                builderObject.seo = {
                   hostname : hostname,
                   locales : rendererObject.locales,
                   products :  rendererObject.products
                }
                path = 'sitemap';
                break;
            case 'main':
                var descr = {
                    'en' : 'Huawei is the world’s largest manufacturer of telecommunication equipment. Visit our site and get all essential information about Huawei smartphones, tablets and hardware. Be the first to know about our new arrivals and hot sales.',
                    'ru' : 'Huawei крупнейший в мире производитель телекоммуникационного оборудования. Получите всю необходимую информацию о Смартфонах, планшетах и других устройствах Huawei. Узнайте первыми о новинках и супер скидках. ',
                    'hy' : 'Huawei  հեռահաղորդակցության սարքավորումների աշխարհի խոշորագույն արտադրող ընկերությունն է: Այցելե՜ք մեր կայք և ստացեք ամբողջական տեղեկատվություն Huawei հեռախոսների, պլանշետների և այլ սարքավորումների մասին: Առաջինը իմացեք մեծ զեղչերի և նոր արտադրանքի մասին:'
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: descr["" + lang],
                    meta_keywords: "Huawei Armenia,Smartphones,Tablets,Huawei,Смартфоны,Планшеты,Хуавей,Телефоны",
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'smartphones':
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: "Welcome to Huawei Armenia official site, here you can find all new products from Huawei.",
                    meta_keywords: "Huawei,main",
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'smartphones';//'main';
                break;
            case 'tablets':
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: "Welcome to Huawei Armenia official site, here you can find all new products from Huawei.",
                    meta_keywords: "Huawei,main",
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'smartphones';//'main';
                break;
            case 'blog':
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Huawei Blog",
                    meta_description: "Welcome to Huawei Armenia official site, here you can find all new products from Huawei.",
                    meta_keywords: "Huawei,main",
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'enterprises':
                builderObject.seo = {
                    title: "Huawei Armenia",
                    meta_description: "description - enterprises",
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    //image: ""
                    //,
                    //lang: req.body.locale
                }
                path = 'enterprises';
                break;
        }
        console.log(__dirname, path, 'botView/' + path + '.html');
        res.render('' + path + '.html', {data: builderObject}, function (error, html) {
            if (!error) {
                res.end(html);
                //res.end();
                return;
            }
            else {
                console.error("renderer error : " + error);
            }
            return;
        });
    }
    catch (e) {
        console.log(e);
    }
}
