// middleware/routes.js
/**
 * Created by mark on 11/12/2015.
 */

module.exports = function (app, fs) {
    app.post('/api/getProducts',function (req, res) {
        var products;
        if(req.body.type == 's') {
            products = [
                {
                    title: "Huawei P6S", description: "", status: ' ', statusText: ' ', img: "/img/main-products/p6s.png", slug: "p6s"
                },
                {
                    title: "Honor 3", description: "", status: ' ', statusText: ' ', img: "/img/main-products/honor-3.png", slug: "honor-3"
                },
                {
                    title: "Huawei G6", description: "", status: ' ', statusText: ' ', img: "/img/main-products/g6-3g.png", slug: "g6-3g"
                },
                {
                    title: "Huawei Y220", description: "", status: ' ', statusText: ' ', img: "/img/main-products/y220.png", slug: "y220"
                },
                {
                    title: "Honor 4C Pro", description: "", status: ' ', statusText: ' ', img: "/img/main-products/4c-pro.png", slug: "4c-pro"
                },
                {
                    title: "Huawei G730", description: "", status: ' ', statusText: ' ', img: "/img/main-products/g730.png", slug: "g730"
                },
                {
                    title: "Huawei P9 Lite", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/p9-lite.png", slug: "p9-lite"
                },
                {
                    title: "Huawei P9", description: "", status: ' ', statusText: ' ', img: "/img/main-products/p9.png", slug: "huawei-p9"
                },
                {
                    title: "Honor 5X", description: "", status: ' ', statusText: ' ', img: "/img/main-products/honor-5x.png", slug: "honor-5x"
                },
                {
                    title: "Huawei Y5 II", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/y5ii.png", slug: "y5ii"
                },
                {
                    title: "Huawei Y3 II", description: "", status: ' ', statusText: ' ', img: "/img/main-products/y3ii.png", slug: "huawei-y3ii"
                },
                {
                    title: "Honor 5C", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/honor-5c.png", slug: "honor-5c"
                },
                {
                    title: "Huawei GR3", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/honor-5c.png", slug: "gr3"
                },
                // {
                //     title: "Honor 8", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/"
                //     //,slug: "honor-8"
                // },
                {
                    title: "Huawei Y6 II", description: "", status: 'new', statusText: 'Coming Soon', img: "/img/main-products/y6ii.png", slug: "y6ii"
                }
            ];
            res.send(products).end();
        }
        else if(req.body.type == 't') {
            products = [
                {
                    title: "Huawei MediaPad X1 7.0", description: "", img: "/img/main-products/mediapad-x1-7-0.png", slug: "mediapad-x1-7"
                },
                {
                    title: "Huawei MediaPad T1 7.0", description: "", img: "/img/main-products/mediapad-t1-7-0.png", slug: "mediapad-t1-7-0"
                },
                {
                    title: "Huawei MediaPad T1 8.0", description: "", img: "/img/main-products/mediapad-t1-8-0.png", slug: "mediapad-t1-8-0"
                }
            ];
            res.send(products).end();
        }
    });
    app.post('/api/getProduct',function (req, res) {
        var path = './data/products/'+ req.body.lang + '-' + req.body.slug + '.json';
        fs.readFile(path, 'utf8', function (err, data) {
            if (err) {
                console.error(err);
                res.send({error : 'No such product.'});
            }
            else {
                var obj = JSON.parse(data);
                res.send(obj);
            }

        });
    });
    app.post('/api/getArticles',function (req, res) {
        var articles;
        switch(req.body.lang) {
            case 'en':
                articles = [
                    {
                        title: "Why It's Impossible not to Love Huawei : 11 Facts", text: "It's sure not easy to choose a smartphone or a gadget these days. Options are vast, criteria different. So it's not easy to cherry-pick and as a result you might end up with a device you bought simply because your friends have it. In order to understand the quality criteria one has to be familiar with industry leaders. Chinese company Huawei is one of the biggest smartphone manufacturers nowadays. A few years back this would have been hard to believe but today it's a stated fact. Last year the company introduced its new concept: В«Make it PossibleВ», and Huawei's new motto is В«Dreams inspire creativityВ». Below are 11 facts about Huawei out of many that will change you perception of this Chinese company.", date: '11.08.2016', img: '1.png', slug: '11_facts_about_huawei'
                    }
                ];
                break;
            case 'ru':
                articles = [
                    {
                        title: "11 С„Р°РєС‚РѕРІ РєРѕС‚РѕСЂС‹Рµ РІС‹ РЅРµР·РЅР°Р»Рё Рѕ Huawei", text: "Р’С‹Р±РѕСЂ СЃРјР°СЂС‚С„РѕРЅРѕРІ Рё РґСЂСѓРіРѕРіРѕ РѕР±РѕСЂСѓРґРѕРІР°РЅРёСЏ РЅР° СЃРµРіРѕРґРЅСЏС€РЅРёР№ РґРµРЅСЊ СЏРІР»СЏРµС‚СЃСЏ РЅРµР»РµРіРєРѕР№ Р·Р°РґР°С‡РµР№. РЈС‡РёС‚С‹РІР°СЏ С€РёСЂРѕРєРёР№ Р°СЃСЃРѕСЂС‚РёРјРµРЅС‚ Рё СЂР°Р·Р»РёС‡РЅС‹Рµ РєСЂРёС‚РµСЂРёРё, РјРѕР¶РЅРѕ Р»РµРіРєРѕ Р·Р°РїСѓС‚Р°С‚СЊСЃСЏ, Рё РїСЂРё РІС‹Р±РѕСЂРµ СЂСѓРєРѕРІРѕРґСЃС‚РІРѕРІР°С‚СЊСЃСЏ РЅРµ РєР°С‡РµСЃС‚РІРѕРј, Р° Р»РёС€СЊ С‚РµРј, С‡С‚Рѕ РІС‹Р±РёСЂР°СЋС‚ РЅР°С€Рё Р·РЅР°РєРѕРјС‹Рµ. Р‘РѕР»РµРµ С‚РѕРіРѕ, РЅРµРѕР±С…РѕРґРёРјРѕ Р·РЅР°С‚СЊ Р»РёРґРµСЂРѕРІ РґР°РЅРЅРѕР№ РѕС‚СЂР°СЃР»Рё РґР»СЏ С‚РѕРіРѕ, С‡С‚РѕР±С‹ РѕРїРµСЂРµРґРµР»РёС‚СЊ РєСЂРёС‚РµСЂРёРё РѕС†РµРЅРєРё РєР°С‡РµСЃС‚РІР°. РљРёС‚Р°Р№СЃРєР°СЏ РєРѕРјРїР°РЅРёСЏ Huawei СЏРІР»СЏРµС‚СЃСЏ РѕРґРЅРёРј РёР· РєСЂСѓРїРЅРµР№С€РёС… РїСЂРѕРёР·РІРѕРґРёС‚РµР»РµР№ СЃРјР°СЂС‚С„РѕРЅРѕРІ. РќРµСЃРєРѕР»СЊРєРѕ Р»РµС‚ РЅР°Р·Р°Рґ РІ СЌС‚Рѕ Р±С‹Р»Рѕ Р±С‹ С‚СЂСѓРґРЅРѕ РїРѕРІРµСЂРёС‚СЊ, РЅРѕ РЅР° СЃРµРіРѕРґРЅСЏС€РЅРёР№ РґРµРЅСЊ СЌС‚Рѕ СЏРІР»СЏРµС‚СЃСЏ РїСЂРёР·РЅР°РЅРЅС‹Рј С„Р°РєС‚РѕРј.", date: '11.08.2016', img: '1.png', slug: '11_facts_about_huawei'
                    }
                ];
                break;
            case 'hy':
            default:
                articles = [
                    {
                        title: "11 ЦѓХЎХЅХї, ХёЦЂХёХ¶Ц„ Х«ХґХЎХ¶ХЎХ¬ХёХѕ Х°Х¶ХЎЦЂХЎХѕХёЦЂ Х№Х§ Х№ХЅХ«ЦЂХҐХ¬ Huawei-ХЁ", text: "ХЌХґХЎЦЂХ©Ц†ХёХ¶Х¶ХҐЦЂ ХҐЦ‚ ХЎХµХ¬ ХїХҐХ­Х¶Х«ХЇХЎ ХЁХ¶ХїЦЂХҐХ¬Х¶ ХЎХµХЅЦ…ЦЂ Х°ХҐХ·Хї ХЈХёЦЂХ® Х№Х§: ФёХ¶ХїЦЂХёЦ‚Х©ХµХёЦ‚Х¶ХЁ ХґХҐХ® Х§,  Х№ХЎЦѓХЎХ¶Х«Х·Х¶ХҐЦЂХЁ` ХїХЎЦЂХўХҐЦЂ, Х«Х¶Х№Х« Х°ХҐХїХҐЦ‚ХЎХ¶Ц„ХёХѕ ХЇХЎЦЂХҐХ¬Х« Х§ Х­ХіХіХѕХҐХ¬ ХёЦ‚ ХёЦЂХЎХЇХ« ЦѓХёХ­ХЎЦЂХҐХ¶ ХЁХ¶ХїЦЂХҐХ¬ ХЁХ¶Х¤ХЎХґХҐХ¶ХЁ ХЎХµХ¶, Х«Х¶Х№ ХЁХ¶ХїЦЂХёЦ‚Хґ ХҐХ¶ ХґХҐЦЂ Х®ХЎХ¶ХёХ©Х¶ХҐЦЂХЁ: Ф»ХЅХЇ ХёЦЂХЎХЇХ« Х№ХЎЦѓХЎХ¶Х«Х·Х¶ХҐЦЂХЁ Х°ХЎХЅХЇХЎХ¶ХЎХ¬ХёЦ‚ Х°ХЎХґХЎЦЂ ХєХҐХїЦ„ Х§ Х®ХЎХ¶ХёХ© Х¬Х«Х¶ХҐХ¬ ХёХ¬ХёЦЂХїХ« ХЎХјХЎХ»ХЎХїХЎЦЂХ¶ХҐЦЂХ«Х¶:", date: '11.08.2016', img: '1.png', slug: '11_facts_about_huawei'
                    }
                ];
                break;
        }

        res.send(articles).end();
    });
    app.post('/api/getArticle',function (req, res) {
        var path = './data/blog/'+ req.body.lang + '-' + req.body.slug;
        fs.readFile(path + '.json', 'utf8', function (err, data) {
            if (err) {
                console.error(err);
                res.send({error : 'No such article.'});
            }
            else {
                var obj = JSON.parse(data);

                fs.readFile(path + '.html', 'utf8', function (err, data1) {
                    if (err) {
                        console.error(err);
                        res.send({error : 'No such article.'});
                    }
                    else {
                        obj.inner = data1;
                        res.send(obj);
                    }

                });
            }

        });
    });
    //app.post('*', function (req, res) {
    //    res.sendFile('index.html', { root: '././files/' });
    //});
    //app.get('*', function (req, res) {
    //    res.sendFile('index.html', { root: '././files/' });
    //});
    app.post('/sitemap.xml', function (req, res) {exports.generateSitemap(req, res)});

    app.get('/sitemap.xml', function (req, res) {exports.generateSitemap(req, res)});

    app.post('/robots.txt', function (req, res) {exports.generateRobots(req, res)});

    app.get('/robots.txt', function (req, res) {exports.generateRobots(req, res)});

    app.get('*', function (req, res) {
        var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
        //exports.botHack(req, res);
        if (userAgent != 'Googlebot' && userAgent != 'bingbot' && userAgent != 'Slurp' && userAgent != 'FacebookBot') {
            res.sendFile('index.html', { root: '././files/' });
        }
        else {
            exports.botHack(req, res);
        }
    });

    app.post('*', function (req, res) {
        var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
        //exports.botHack(req, res);
        if (userAgent != 'Googlebot' && userAgent != 'bingbot' && userAgent != 'Slurp' && userAgent.indexOf('Facebook') == -1 && userAgent.indexOf('facebookexternalhit/1.1') == -1 && userAgent != 'Facebot') {
            res.sendFile('index.html', { root: '././files/' });
        }
        else {
            exports.botHack(req, res);
        }
    });
};


exports.botHack = function (req, res) {
    var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
    var renderObject = {
        test : "test-dev"
    };
    try {
        if (req.url == '') {
            console.log("bot / main");
            req.route.path = 'main';
        }
        else if (req.url == '/') {
            console.log("bot / main");
            req.route.path = 'main';
        }
        else if (req.url.indexOf('blog') > -1) {
            console.log("bot / blog");
            req.route.path = 'blog';
        }
        else if (req.url.indexOf('product') > -1) {
            console.log("bot / product");
            req.route.path = 'product';
            req.route.slug = req.url.split('products/')[1];
        }
        else if (req.url.indexOf('enterprises') > -1) {
            console.log("bot / enterprises");
            req.route.path = 'enterprices';
        }
        else if (req.url.indexOf('aboutus') > -1) {
            console.log("bot / aboutus");
            req.route.path = 'aboutus';
        }
        else if (req.url.indexOf('termsofuse') > -1) {
        }
        else if (req.url.indexOf('privacy') > -1) {
        }
        else if (req.url.indexOf('signup') > -1) {
        }
        else if (req.url.indexOf('genre') > -1) {
        }
        else if (req.url.indexOf('article') > -1) {
        }
        else {
            req.route.path = 'main';
            console.log("bot / other");
        }
        var routeName = req.route.path.indexOf(':') === -1 ? req.route.path : req.route.path.substring(0, req.route.path.indexOf(":") - 1);
        console.log(routeName);
        exports.renderer(req, res, renderObject, routeName);
    }
    catch (e) {
        console.log({'level': 'ERROR', 'message': e});
    }
}

exports.generateSitemap = function(req, res) {
    var userAgent = uaParser.parse(req.headers['user-agent']).userAgent.family;
    var renderObject = {
        locales : ['hy','en','ru'],
        products : [
            {title:"Huawei NEXUS 6P",
                description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"},
            {title:"Huawei NEXUS 6P",description:"5.7\" Single SIM smartphone with AMOLED displayy",
                img:"/img/main-products/Huawei NEXUS 6P.png",slug:"huawei-nexus-6p"}]
    };
    try {
        exports.renderer(req, res, renderObject, 'sitemap');
    }
    catch (e) {
        console.log({'level': 'ERROR', 'message': e});
    }
}

exports.renderer = function (req, res, rendererObject, from) {
    var hostname = req.headers.host.indexOf(':') === -1 ? req.headers.host : req.headers.host.substring(0, req.headers.host.indexOf(':'));
    try {
        var lang;
        if(req.url.slice(3,4)=="/") lang = req.url.slice(1,3); else lang='hy';
        var builderObject = {}, path, platformData = {}, lang_links=[];
        var metaDesc = {
            'en' : 'Huawei is the worldвЂ™s largest manufacturer of telecommunication equipment. Visit our site and get all essential information about Huawei smartphones, tablets and hardware. Be the first to know about our new arrivals and hot sales. ',
            'ru' : 'Huawei РєСЂСѓРїРЅРµР№С€РёР№ РІ РјРёСЂРµ РїСЂРѕРёР·РІРѕРґРёС‚РµР»СЊ С‚РµР»РµРєРѕРјРјСѓРЅРёРєР°С†РёРѕРЅРЅРѕРіРѕ РѕР±РѕСЂСѓРґРѕРІР°РЅРёСЏ. РџРѕР»СѓС‡РёС‚Рµ РІСЃСЋ РЅРµРѕР±С…РѕРґРёРјСѓСЋ РёРЅС„РѕСЂРјР°С†РёСЋ Рѕ РЎРјР°СЂС‚С„РѕРЅР°С…, РїР»Р°РЅС€РµС‚Р°С… Рё РґСЂСѓРіРёС… СѓСЃС‚СЂРѕР№СЃС‚РІР°С… Huawei. РЈР·РЅР°Р№С‚Рµ РїРµСЂРІС‹РјРё Рѕ РЅРѕРІРёРЅРєР°С… Рё СЃСѓРїРµСЂ СЃРєРёРґРєР°С….',
            'hy' : 'Huawei  Х°ХҐХјХЎХ°ХЎХІХёЦЂХ¤ХЎХЇЦЃХёЦ‚Х©ХµХЎХ¶ ХЅХЎЦЂЦ„ХЎХѕХёЦЂХёЦ‚ХґХ¶ХҐЦЂХ« ХЎХ·Х­ХЎЦЂХ°Х« Х­ХёХ·ХёЦЂХЎХЈХёЦ‚ХµХ¶ ХЎЦЂХїХЎХ¤ЦЂХёХІ ХЁХ¶ХЇХҐЦЂХёЦ‚Х©ХµХёЦ‚Х¶Х¶ Х§: Ф±ХµЦЃХҐХ¬ХҐХњЦ„ ХґХҐЦЂ ХЇХЎХµЦ„ Ц‡ ХЅХїХЎЦЃХҐЦ„ ХЎХґХўХёХІХ»ХЎХЇХЎХ¶ ХїХҐХІХҐХЇХЎХїХѕХёЦ‚Х©ХµХёЦ‚Х¶ Huawei Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂХ«, ХєХ¬ХЎХ¶Х·ХҐХїХ¶ХҐЦЂХ« Ц‡ ХЎХµХ¬ ХЅХЎЦЂЦ„ХЎХѕХёЦЂХёЦ‚ХґХ¶ХҐЦЂХ« ХґХЎХЅХ«Х¶: Ф±ХјХЎХ»Х«Х¶ХЁ Х«ХґХЎЦЃХҐЦ„ ХґХҐХ® Х¦ХҐХІХ№ХҐЦЂХ« Ц‡ Х¶ХёЦЂ ХЎЦЂХїХЎХ¤ЦЂХЎХ¶Ц„Х« ХґХЎХЅХ«Х¶:'
        }
        switch (from) {
            case 'robots':
                builderObject.seo = {
                    hostname : hostname
                }
                path = 'robots';
                break;
            case 'sitemap':
                builderObject.seo = {
                   hostname : hostname,
                   locales : rendererObject.locales,
                   products :  rendererObject.products
                }
                path = 'sitemap';
                break;
            case 'main':
                var descr = {
                    'en' : 'Huawei is the worldвЂ™s largest manufacturer of telecommunication equipment. Visit our site and get all essential information about Huawei smartphones, tablets and hardware. Be the first to know about our new arrivals and hot sales.',
                    'ru' : 'Huawei РєСЂСѓРїРЅРµР№С€РёР№ РІ РјРёСЂРµ РїСЂРѕРёР·РІРѕРґРёС‚РµР»СЊ С‚РµР»РµРєРѕРјРјСѓРЅРёРєР°С†РёРѕРЅРЅРѕРіРѕ РѕР±РѕСЂСѓРґРѕРІР°РЅРёСЏ. РџРѕР»СѓС‡РёС‚Рµ РІСЃСЋ РЅРµРѕР±С…РѕРґРёРјСѓСЋ РёРЅС„РѕСЂРјР°С†РёСЋ Рѕ РЎРјР°СЂС‚С„РѕРЅР°С…, РїР»Р°РЅС€РµС‚Р°С… Рё РґСЂСѓРіРёС… СѓСЃС‚СЂРѕР№СЃС‚РІР°С… Huawei. РЈР·РЅР°Р№С‚Рµ РїРµСЂРІС‹РјРё Рѕ РЅРѕРІРёРЅРєР°С… Рё СЃСѓРїРµСЂ СЃРєРёРґРєР°С…. ',
                    'hy' : 'Huawei  Х°ХҐХјХЎХ°ХЎХІХёЦЂХ¤ХЎХЇЦЃХёЦ‚Х©ХµХЎХ¶ ХЅХЎЦЂЦ„ХЎХѕХёЦЂХёЦ‚ХґХ¶ХҐЦЂХ« ХЎХ·Х­ХЎЦЂХ°Х« Х­ХёХ·ХёЦЂХЎХЈХёЦ‚ХµХ¶ ХЎЦЂХїХЎХ¤ЦЂХёХІ ХЁХ¶ХЇХҐЦЂХёЦ‚Х©ХµХёЦ‚Х¶Х¶ Х§: Ф±ХµЦЃХҐХ¬ХҐХњЦ„ ХґХҐЦЂ ХЇХЎХµЦ„ Ц‡ ХЅХїХЎЦЃХҐЦ„ ХЎХґХўХёХІХ»ХЎХЇХЎХ¶ ХїХҐХІХҐХЇХЎХїХѕХёЦ‚Х©ХµХёЦ‚Х¶ Huawei Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂХ«, ХєХ¬ХЎХ¶Х·ХҐХїХ¶ХҐЦЂХ« Ц‡ ХЎХµХ¬ ХЅХЎЦЂЦ„ХЎХѕХёЦЂХёЦ‚ХґХ¶ХҐЦЂХ« ХґХЎХЅХ«Х¶: Ф±ХјХЎХ»Х«Х¶ХЁ Х«ХґХЎЦЃХҐЦ„ ХґХҐХ® Х¦ХҐХІХ№ХҐЦЂХ« Ц‡ Х¶ХёЦЂ ХЎЦЂХїХЎХ¤ЦЂХЎХ¶Ц„Х« ХґХЎХЅХ«Х¶:'
                };
                var keywords = {
                    "en" : "huawei,huawei phones,huawei mobile,huawei honor,huawei tablet,smart phones,huawei smartphone,telephone huawei,huawei telefon,huawei 8,huawei mobile phones,huawei technologies,huawei device,huawei mate 2,huawei shop,best huawei phone,huawei android,huawei latest phone,huawei new phone,huawei review,huawei company,huawei website",
                    "ru" : "РҐСѓР°РІРµР№,РҐСѓР°РІРµР№ С‚РµР»РµС„РѕРЅС‹,РҐСѓР°РІРµР№ РјРѕР±РёР»СЊРЅРёРєРё,РҐСѓР°РІРµР№ С…РѕРЅРѕСЂ,С…СѓР°РІРµР№ РїР»Р°РЅС€РµС‚С‹,РїР»Р°РЅС€РµС‚С‹,РҐСѓР°РІРµР№ СЃРјР°СЂС‚С„РѕРЅС‹,СЃРјР°СЂС‚С„РѕРЅС‹,С‚РµР»РµС„РѕРЅС‹ С…СѓР°РІРµР№,С‚РµР»РµС„РѕРЅС‹ С…СѓР°РІРµР№,С…СѓР°РІРµР№ 8,РјРѕР±РёР»СЊРЅС‹Рµ С‚РµР»РµС„РѕРЅС‹,С…СѓР°РІРµР№ С‚РµС…РЅРѕР»РѕРіРёСЏ,С…СѓР°РІРµР№ РїСЂРѕРґСѓРєС†РёСЏ,huawei mate 2,С…СѓР°РІРµР№ РјР°РіР°Р·РёРЅ,Р»СѓС‡С€Р°СЏ С†РµРЅР° С…СѓР°РІРµР№,Р°РЅРґСЂРѕРёРґ С…СѓР°РІРµР№,РїРѕСЃР»РµРґРЅРёРµ РјРѕРґРµР»Рё,РЅРѕРІРёРЅРєРё,РѕС‚Р·С‹РІС‹,РєРѕРјРїР°РЅРёСЏ,РІРµР± СЃР°Р№С‚,С…СѓР°РІРµР№,РҐСѓР°РІРµР№ РђСЂРјРµРЅРёСЏ,РҐСѓР°РІРµР№ Р•СЂРµРІР°РЅ",
                    "hy" : "ФЅХёЦ‚ХЎХѕХҐХµ,ХЂХёЦ‚ХЎХѕХҐХµ,huawei,huawei ХЂХҐХјХЎХ­ХёХЅ,huawei Х„ХёХўХЎХµХ¬,huawei honor,huawei ХїХЎХўХ¬ХҐХї, ХЅХґХЎЦЂХїЦ†ХёХ¶Х¶ХҐЦЂ,huawei ХЅХґХЎЦЂХїЦ†ХёХ¶Х¶ХҐЦЂ, Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ huawei,huawei 8,huawei ХґХёХўХЎХµХ¬ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei ХїХҐХ­Х¶ХёХ¬ХёХЈХ«ХЎ, huawei ХЅХЎЦЂЦ„ХҐЦЂ,huawei mate 2, huawei Х­ХЎХ¶ХёЦ‚Х©,ФјХЎХѕХЎХЈХёЦ‚ХµХ¶ huawei Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei android,huawei ХѕХҐЦЂХ»Х«Х¶ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei Х¶ХёЦЂ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei review,huawei ХЁХ¶ХЇХҐЦЂХёЦ‚Х©ХµХёЦ‚Х¶, huawei ХЇХЎХµЦ„,ФіХ¶ХҐХ¬ huawei,Ф±ХјХ¶ХҐХ¬ huawei"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'smartphones':
                var keywords = {
                    "en" : "huawei phones,huawei mobile,huawei honor,huawei,smart phones,huawei smartphone,huawei ascend,telephone huawei,huawei telefon,huawei 8,huawei mobile phones,huawei phone price,huawei mobile price,huawei device,huawei mate 2,mediapad,huawei e5220,huawei honor 5c,honor huawei,huawei shop,huawei phones price list,best huawei phone,huawei android,huawei price,huawei latest phone,huawei new phone,huawei p8 lite,nexus 6p,huawei mate 8,huawei mate s,huawei mate 7,huawei ascend p7,huawei ascend mate 7,huawei honor 6,huawei g elite,telephone huawei,Huawei Armenia,Huawei Yerevan",
                    "ru" : "РҐСѓР°РІРµР№ С‚РµР»РµС„РѕРЅС‹,РҐСѓР°РІРµР№ С…РѕРЅРѕСЂ,РҐСѓР°РІРµР№,СЃРјР°СЂС‚С„РѕРЅС‹,С…СѓР°РІРµР№ СЃРјР°СЂС‚С„РѕРЅС‹,huawei ascend,С‚РµР»РµС„РѕРЅС‹ С…СѓР°РІРµР№,РҐСѓР°РІРµР№ С‚РµР»РµС„РѕРЅС‹,С…СѓР°РІРµР№ 8,РјРѕР±РёР»СЊРЅС‹Рµ С‚РµР»РµС„РѕРЅС‹,С…СѓР°РІРµР№ С†РµРЅР°,С…СѓРІР°РµР№ СЃРјР°СЂС‚С„РѕРЅС‹ С†РµРЅР°,С…СѓР°РІРµР№ РїСЂРѕРґСѓРєС†РёСЏ,huawei mate 2,mediapad,huawei e5220,huawei honor 5c,honor huawei,huawei shop,huawei phones price list,Р»СѓС‡С€Р°СЏ С†РµРЅР° С…СѓР°РІРµР№,Р°РЅРґСЂРѕРёРґ С…СѓР°РІРµР№,С…СѓР°РІРµР№ С†РµРЅР°,РїРѕСЃР»РµРґРЅРёРµ РјРѕРґРµР»Рё,РЅРѕРІРёРЅРєРё,huawei p8 lite,nexus 6p,huawei mate 8,huawei mate s,huawei mate 7,huawei ascend p7,huawei ascend mate 7,huawei honor 6,huawei g elite,РҐСѓР°РІРµР№ РђСЂРјРµРЅРёСЏ,РҐСѓР°РІРµР№ Р•СЂРµРІР°РЅ",
                    "hy" : "huawei ХЂХҐХјХЎХ­ХёХЅ,huawei Х„ХёХўХЎХµХ¬,huawei honor,huawei,ХЅХґХЎЦЂХїЦ†ХёХ¶Х¶ХҐЦЂ,huawei ХЅХґХЎЦЂХїЦ†ХёХ¶Х¶ХҐЦЂ,huawei ascend,Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ huawei,huawei 8,huawei ХґХёХўХЎХµХ¬ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei Х°ХҐХјХЎХ­ХёХЅХ« ХЈХ¶ХҐЦЂ,huawei ХґХёХўХЎХµХ¬ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei ХЅХЎЦЂЦ„ХҐЦЂ,huawei mate 2,mediapad,huawei e5220,huawei honor 5c,honor huawei,huawei Х­ХЎХ¶ХёЦ‚Х©,huawei Х°ХҐХјХЎХ­ХёХЅХ« ХЈХ¶ХҐЦЂ,ФјХЎХѕХЎХЈХёЦ‚ХµХ¶ huawei Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei android,huawei ХЈХ¶ХҐЦЂ,huawei ХѕХҐЦЂХ»Х«Х¶ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei Х¶ХёЦЂ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei p8 lite,nexus 6p,huawei mate 8,huawei mate s,huawei mate 7,huawei ascend p7,huawei ascend mate 7,huawei honor 6,huawei g elite,telephone huawei,Huawei ХЂХЎХµХЎХЅХїХЎХ¶,Huawei ФµЦЂЦ‡ХЎХ¶,huawei ХЇХЎЦЂХ®Х«Ц„Х¶ХҐЦЂ,huawei ХЇХЎЦЂХ®Х«Ц„,huawei ХЎЦЂХїХЎХ¤ЦЂХЎХ¶Ц„,ФіХ¶ХҐХ¬ huawei,Ф±ХјХ¶ХҐХ¬ huawei"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'tablets':
                var keywords = {
                    "en" : "huawei,huawei tablet,huawei mediapad,huawei technologies,huawei device,huawei tablet price,mediapad,huawei tablet price list,huawei shop,Huawei Armenia,Huawei Yerevan,huawei review",
                    "ru" : "РҐСѓР°РІРµР№,РїР»Р°РЅС€РµС‚С‹,huawei mediapad,С…СѓР°РІРµР№ С‚РµС…РЅРѕР»РѕРіРёСЏ,С…СѓР°РІРµР№ РїСЂРѕРґСѓРєС†РёСЏ,С…СѓР°РІРµР№ РїР»Р°РЅС€РµС‚С‹ С†РµРЅР°,mediapad,РѕС‚Р·С‹РІС‹,С…СѓР°РІРµР№ РјР°РіР°Р·РёРЅ,Huawei Armenia,Huawei Yerevan,huawei review,РҐСѓР°РІРµР№ РђСЂРјРµРЅРёСЏ,РҐСѓР°РІРµР№ Р•СЂРµРІР°РЅ",
                    "hy" : "huawei,huawei ХїХЎХўХ¬ХҐХї,huawei mediapad,huawei ХїХҐХ­Х¶ХёХ¬ХёХЈХ«ХЎ,huawei device,huawei ХєХ¬ХЎХ¶Х·ХҐХї ХЈХ«Х¶,mediapad,huawei ХїХЎХўХ¬ХҐХї ХЈХ«Х¶,huawei Х­ХЎХ¶ХёЦ‚Х©,Huawei ХЂХЎХµХЎХЅХїХЎХ¶,Huawei ФµЦЂЦ‡ХЎХ¶,huawei ХЇХЎЦЂХ®Х«Ц„,ФЅХёЦ‚ХЎХѕХҐХµ,ХЂХёЦ‚ХЎХѕХҐХµ,Huawei ХєХ¬ХЎХ¶Х·ХҐХїХ¶ХҐЦЂ,Huawei ХєХ¬ХЎХ¶Х·ХҐХїХ¶ХҐЦЂХ« ХЈХ¶ХҐЦЂ,huawei Х­ХЎХ¶ХёЦ‚Х©,ФіХ¶ХҐХ¬ huawei,Ф±ХјХ¶ХҐХ¬ huawei"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Official Site",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'blog':
                var keywords = {
                    "en" : "huawei phones,huawei mobile,huawei tablet,smart phones,Huawei Armenia,Huawei Yerevan",
                    "ru" : "РҐСѓР°РІРµР№ С‚РµР»РµС„РѕРЅС‹,С…СѓР°РІРµР№ РјР°РіР°Р·РёРЅ,С…СѓР°РІРµР№ РїР»Р°РЅС€РµС‚С‹,smart phones,Huawei РђСЂРјРµРЅРёСЏ,Huawei Р•СЂРµРІР°РЅ",
                    "hy" : "huawei ХЂХҐХјХЎХ­ХёХЅ,huawei Х„ХёХўХЎХµХ¬,Huawei ХєХ¬ХЎХ¶Х·ХҐХїХ¶ХҐЦЂ,huawei ХЅХґХЎЦЂХїЦ†ХёХ¶Х¶ХҐЦЂ,Huawei ХЂХЎХµХЎХЅХїХЎХ¶,Huawei ФµЦЂЦ‡ХЎХ¶,ФЅХёЦ‚ХЎХѕХҐХµ,ХЂХёЦ‚ХЎХѕХҐХµ,huawei ХѕХҐЦЂХ»Х«Х¶ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,huawei Х¶ХёЦЂ Х°ХҐХјХЎХ­ХёХЅХ¶ХҐЦЂ,Huawei Х¶ХёЦЂХёЦ‚Х©ХµХёЦ‚Х¶Х¶ХҐЦЂ,Huawei ХєХ¬ХЎХ¶Х·ХҐХїХ¶ХҐЦЂ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Huawei Blog",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';//'main';
                break;
            case 'enterprises':
                var keywords = {
                    "en" : "huawei,huawei phones,huawei device,huawei technologies co ltd,Huawei become a partner,Huawei partner,Huawei Mobile Phones partiler,huawei shop,Huawei Armenia,Huawei Yerevan,huawei review",
                    "ru" : "РҐСѓР°РІРµР№,РҐСѓР°РІРµР№ С‚РµР»РµС„РѕРЅС‹,С…СѓР°РІРµР№ СѓСЃС‚СЂРѕР№СЃС‚РІР°,С…СѓР°РІРµР№ С‚РµС…РЅРѕР»РѕРіРёРё,Huawei РґР»СЏ РїР°СЂС‚РЅРµСЂРѕРІ,Huawei РїР°СЂС‚РЅРµСЂСЃС‚РІРѕ,Huawei РјРѕР±РёР»СЊРЅС‹Рµ СѓСЃС‚СЂРѕР№СЃС‚РІР°,С…СѓР°РІРµР№ РјР°РіР°Р·РёРЅ,Huawei РђСЂРјРµРЅРёСЏ,Huawei Р•СЂРµРІР°РЅ,huawei РѕС‚Р·С‹РІС‹",
                    "hy" : "huawei,huawei ХЂХҐХјХЎХ­ХёХЅ,huawei ХЅХЎЦЂЦ„ХҐЦЂ,huawei ХЎЦЂХїХЎХ¤ЦЂХЎХ¶Ц„,huawei ХїХЎХўХ¬ХҐХї,Huawei Х¤ХЎЦЂХ±Х«ЦЂ ХЈХёЦЂХ®ХЁХ¶ХЇХҐЦЂ,Huawei ХЈХёЦЂХ®ХЁХ¶ХЇХҐЦЂ,huawei ХЅХґХЎЦЂХїЦ†ХёХ¶Х¶ХҐЦЂ,huawei Х­ХЎХ¶ХёЦ‚Х©,Huawei ХЂХЎХµХЎХЅХїХЎХ¶,Huawei ФµЦЂЦ‡ХЎХ¶,ФЅХёЦ‚ХЎХѕХҐХµ,ХЂХёЦ‚ХЎХѕХҐХµ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Enterprises",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'aboutus':
                var keywords = {
                    "en" : "huawei,contacts,contact us,huawei mobile,huawei tablet,huawei shop,huawei technologies co ltd,Huawei Armenia,Huawei Yerevan,huawei company,about Huawei,about us",
                    "ru" : "РҐСѓР°РІРµР№,РєРѕРЅС‚Р°РєС‚С‹,СЃРІСЏР¶РёС‚РµСЃСЊ СЃ РЅР°РјРё,РҐСѓР°РІРµР№ С‚РµР»РµС„РѕРЅС‹,С…СѓР°РІРµР№ РїР»Р°РЅС€РµС‚С‹,С…СѓР°РІРµР№ РјР°РіР°Р·РёРЅ,С…СѓР°РІРµР№ СѓСЃС‚СЂРѕР№СЃС‚РІР°,Huawei РђСЂРјРµРЅРёСЏ,Huawei Р•СЂРµРІР°РЅ,РєРѕРјРїР°РЅРёСЏ Huawei,РїСЂРѕ РєРѕРјРїР°РЅРёСЋ Huawei,Рѕ РЅР°СЃ",
                    "hy" : "ХґХҐЦЂ ХґХЎХЅХ«Х¶,ХЇХЎХє,Huawei ХЁХ¶ХЇХҐЦЂХёЦ‚Х©ХµХёЦ‚Х¶,Huawei Х¶ХҐЦЂХЇХЎХµХЎЦЃХёЦ‚ЦЃХ«Х№,huawei Х­ХЎХ¶ХёЦ‚Х©,huawei Х­ХЎХ¶ХёЦ‚Х©,Huawei ХґХЎХЅХ«Х¶,Huawei ХЂХЎХµХЎХЅХїХЎХ¶,Huawei ФµЦЂЦ‡ХЎХ¶,ФЅХёЦ‚ХЎХѕХҐХµ,ХЂХёЦ‚ХЎХѕХҐХµ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "About",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'ourpartners':
                var keywords = {
                    "en" : "Huawei become a partner,Huawei partner,Huawei Mobile Phones partiler,huawei shop,Huawei Armenia,Huawei Yerevan,huawei review",
                    "ru" : "Huawei СЃС‚Р°РЅСЊ РїР°СЂС‚РЅРµСЂРѕРј,Huawei РїР°СЂС‚РЅРµСЂС‹,Huawei Mobile Phones partiler,huawei shop,Huawei РђСЂРјРµРЅРёСЏ,Huawei Р•СЂРµРІР°РЅ,huawei РѕС‚Р·С‹РІС‹,Huawei СЃС‚Р°РЅСЊ РїСЂРµРґСЃС‚Р°РІРёС‚РµР»РµРј",
                    "hy" : "Huawei Х¤ХЎЦЂХ±Х«ЦЂ ХЈХёЦЂХ®ХЁХ¶ХЇХҐЦЂ,Huawei ХЈХёЦЂХ®ХЁХ¶ХЇХҐЦЂ,Huawei Mobile Phones partiler,huawei Х­ХЎХ¶ХёЦ‚Х©,huawei ХЇХЎЦЂХ®Х«Ц„,Huawei ХЂХЎХµХЎХЅХїХЎХ¶,Huawei ФµЦЂЦ‡ХЎХ¶,ФЅХёЦ‚ХЎХѕХҐХµ,ХЂХёЦ‚ХЎХѕХҐХµ,ХЈХёЦЂХ®ХЁХ¶ХЇХҐЦЂХ¶ХҐЦЂ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Partners",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'contacts':
                var keywords = {
                    "en" : "huawei,contacts,contact us,huawei mobile,huawei tablet,huawei shop,huawei technologies co ltd,Huawei Armenia,Huawei Yerevan,huawei company,Huawei address,Huawei ardess,Huawei Map,Huawei number,Huawei office",
                    "ru" : "huawei,РєРѕРЅС‚Р°РєС‚С‹,СЃРІСЏР¶РёС‚РµСЃСЊ СЃ РЅР°РјРё,РҐСѓР°РІРµР№ С‚РµР»РµС„РѕРЅС‹,С…СѓР°РІРµР№ РїР»Р°РЅС€РµС‚С‹,С…СѓР°РІРµР№ РјР°РіР°Р·РёРЅ,С…СѓР°РІРµР№ СѓСЃС‚СЂРѕР№СЃС‚РІР°,Huawei РђСЂРјРµРЅРёСЏ,Huawei Р•СЂРµРІР°РЅ,РєРѕРјРїР°РЅРёСЏ Huawei,Huawei Р°РґСЂРµСЃСЃ,Huawei Р°РґСЂРµСЃСЃ РјР°РіР°Р·РёРЅР°,Huawei РљР°СЂС‚Р°,Huawei РЅРѕРјРµСЂ С‚РµР»РµС„РѕРЅР°,Huawei РѕС„РёСЃ",
                    "hy" : "Х°ХЎХЅЦЃХҐ,ХЇХЎХє,Х°ХҐХјХЎХ­ХёХЅ,huawei ХЁХ¶ХЇХҐЦЂХёЦ‚Х©ХµХёЦ‚Х¶,Huawei Х°ХЎХЅЦЃХҐ,huawei Х­ХЎХ¶ХёЦ‚Х©,Huawei Х­ХЎХ¶ХёЦ‚Х©,Huawei Х°ХҐХјХЎХ­ХёХЅХЎХ°ХЎХґХЎЦЂ,Huawei ХЈЦЂХЎХЅХҐХ¶ХµХЎХЇ,Huawei ХЂХЎХµХЎХЅХїХЎХ¶,Huawei ФµЦЂЦ‡ХЎХ¶,ФЅХёЦ‚ХЎХѕХҐХµ,ХЂХёЦ‚ХЎХѕХҐХµ,huawei ХЂХҐХјХЎХ­ХёХЅ"
                };
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: "Contacts",
                    meta_description: metaDesc["" + lang],
                    meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'main';
                break;
            case 'product':
                var productS = {
                    "p6s": {
                        title: "Huawei P6S",
                        description: {
                            hy : "???????? ??????????????????: Huawei P6S-? ??????????? ? ???? ???????? ?????? ????????? ?????? ????? ?????????? Huawei P6 ?????? ????????: ? ????????????? ???????, P6S ?????? ???? 2 SIM-????, ??? 1,6 ??? ????????, ????? 16 ?? ????????? ?????? ????????????, Android 4.4 ????????? ???????? ? ???????? Emotion UI 2.3 ?????????:",
                            ru : "Возвращение совершенства. Huawei Ascend P6S — это развитие успешной модели Huawei Ascend P6 — самого тонкого смартфона в мире (2013). Ascend P6S дополняет предшественника поддержкой двух SIM карт, новым процессором 1,6 ГГц, увеличенным объемом внутренней памяти до 16 Гб, обновленной версией Android 4.4 и фирменным интерфейсом Emotion UI 2.3.",
                            en : "Back to perfection. Huawei Ascend P6S — is the development of the successful model Huawei Ascend P6— the thinnest smartphone in the world (2013). Ascend P6S supplements its previous model with 2 SIM cards, new 1,6 GHz processor, expanded volume of internal memory up to 16 GB, updated version of Android 4.4 and Emotion UI 2.3 interface."
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.jpg",
                        slug: "p6s"
                    },
                    "honor-3": {
                        title: "Honor 3",
                        description: {
                            en : "Ultra Elegant, compact design. Huawei Honor 3 has an elegant design and is made from high-quality plastics. Despite the large screen size, the phone is comfortable to hold in hand.Water and Dust resistance. Huawei Honor 3 is certified according to IP57 standard for protection from dust and water. The phone survives up to 1m immertion in water for no more than 30 minutes.",
                            ru : "Элегантный компактный дизайн. Huawei Honor 3 выполнен в элегантном стиле из качественного пластика. Несмотря на большую диагональ экрана, телефон удобно размещается в руке.",
                            hy : "???????, ??????? ??????: Huawei Honor 3 ????????? ?????????? ? ??????? ??? ????????? ?????????: ?????? ?????? ??? ?????????? ???????? ?????? ? ????? ???????:"
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.jpg",
                        slug: "honor-3"
                    },
                    "g6-3g": {
                        title: "Huawei G6",
                        description: {
                            hy : "????? ???????????: ??????? ? ??????? ??????????????? ??? ?????????? ??????? 42 ????/? ??????????? 3G ?????????:",
                            ru : "Высокая скорость.Подключайтесь и мгновенно делитесь информацией на скорости до 42 Мб/с в сетях 3G.",
                            en : "It’s a Beautiful Thing. Take one look at the Ascend G6 3G and you’ll immediately notice what sets it apart from the rest. Its unbelievably slim 7.85 mm profile, stylish shape and elegantly curved edges take smartphone design to a new level of perfection."
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.jpg",
                        slug: "g6-3g"
                    },
                    "y220": {
                        title: "Huawei Y220",
                        description: {
                            hy : "??? ?????: 3.5 ????????? ????? ????????????? ?????? ??????? Y220 ????????? ??? ????????? ??? ????????? ??????? ????????:",
                            ru : "Большой экран. Ascend Y220 – это смартфон, работа с которым предоставит Вам необычайное удовольствие благодаря возможностям 3.5” экрана высокого разрешения.",
                            en : "Big screen entertainment. In your palm. Get sucked in. The Ascend Y220 is the smartphone that gives you an incredible visual experience you can carry with you every day. Enjoy the ultimate, performance driven graphic experience with a 3.5 inch, high-resolution, multi-touch screen. The next evolution of visual technology is here right in your hand."
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.jpg",
                        slug: "y220"
                    },
                    "4c-pro": {
                        title: "Honor 4C Pro",
                        description: {
                            hy : "??????????? ?????? ? ??????????? ?????????????: ????? ?????? ? ????? ??????, ??? ?????? ???????? ????????? ? ???? ??????? ????: ???? ?????? ?????????? ???????? ????????? ? ?????? ???????:",
                            ru : "4000 mAh battery. Shorter charging time: 3hrs talktime with 10 minutes charge. With SmartPower 3.0, you can go the extra distance by saving over 30% of energy. Go for 20hrs with just 10% battery. The smartphone tracks background applications that consume energy and offers offers ways to optimize battery life. Charge another phone. 48 hours standby, moderate use.",
                            en : "Новаторский дизайн и современные технологии. Тонкий и эргономичный корпус. Специальная обработка, внешне и тактильно создающая эффект металлического корпуса. Задняя панель с зеркальным эффектом создаёт текстуру, изменяющуюся в зависимости от того, как падает свет."
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.png",
                        slug: "4c-pro"
                    },
                    "g730": {
                        title: "Huawei G730",
                        description: {
                            hy : "????? ?? ????: G730 ????????? ???? 2 ????????, ???? ????? ??? ??? ?????? ????? ?????? ??????????????????: 5?? ????????? ??????? ???? ??????? ????? ?????? ???????? ???? ????????????????? ????????????: ?????? ????????????, ???????? ??? “face detection” , “tracking”, “ smile shutter” ? “facial beautification”, ????? ???? ??? ?????? ????? ??????????? ????????????:",
                            ru : "Большой экран. 14 см (5.5”) qHD экран смартфона Ascend G730 покажет все до мельчайших деталей. Он поддерживает одновременное касание 5 пальцев, что выводит работу с ним на совершенно новый уровень.",
                            en : "A Larger Than Life Experience. The 14 cm (5.5 inch) qHD screen of Ascend G730 brings every detail of your world to life. Be it the excitement of a car race or the life like pictures of your loved ones, the large screen of the Ascend G730 shows everything in vivid detail. This screen comes with the 5 point touch feature that takes the user experience to a whole new level.",
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.jpg",
                        slug: "g730"
                    },
                    "p9-lite": {
                        title: "Huawei P9 Lite",
                        description: {
                            hy : "??? ??: ??? ?????????????? ?????????????? ?????????? Huawei P9 lite-? P8 lite ?????? ???????? ? ? ?????????? ? ?? ??????:",
                            ru : "Новый стиль. Воплощение новых технологий в смартфоне Huawei P9 lite, преемнике P8 lite: лучшие в своем классе решения и самые современые технологии в компактном, стильном корпусе.",
                            en : "HUAWEI P9lite. Turn the dreams of tomorrow into today’s reality with the Huawei P9 lite, successor of the P8 lite. It features a number of innovative hardware upgrades showcased in a cutting-edge slim design. Take hold of a next generation smart device that combines dynamic design with breathtaking style in perfect harmony, all at a competitive price."
                        },
                        status: 'new',
                        statusText: 'Coming Soon',
                        img: "1.jpg",
                        slug: "p9-lite"
                    },
                    "huawei-p9": {
                        title: "Huawei P9",
                        description: {
                            hy : "???????? ??????????????????` dual-????????: Huawei-? ? Leica-? ??????? ?? ????????? dual-????????` ?????????? ????? ??? ????? ? ????? ??? ?????????:",
                            ru : "НОВАЯ ЭРА МОБИЛЬНОЙ ФОТОГРАФИИ. Сочетание качественных материалов, программного обеспечения и двухвекового опыта создания фототехники линз Leica делают Huawei P9 невероятно компактным инструментом для создания фотографий высокого качества.",
                            en : "REINVENT SMARTPHONE PHOTOGRAPHY. Blending superior hardware and software with Leica dual lenses, the Huawei P9 enables you to capture high quality pictures without compromising on the handset’s sleek and compact style."
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.jpg",
                        slug: "huawei-p9"
                    },
                    "honor-5x": {
                        title: "Honor 5X",
                        description: {
                            hy : "2 ?? ???????? ???????????? ?? 8-????????? QUALCOMM ????????: ???? ? ??????? ????????: 2 ?? ???????? ????????????, 16 ?? ?????? ????????????: Qualcomm Snapdragon ???????? ? big.LITTLE ??????????? ????????? ?? ????? ??????????????????? ? ???????????????????: 3D ?????? ? ????????? ??? ??????? ???????? ???????? ???? ?????????? ?????????? ????????? ???????:",
                            ru : "2 ГБ ОПЕРАТИВНОЙ ПАМЯТИ И 8-ЯДЕРНЫЙ ПРОЦЕССОР QUALCOMM. Быстрая и бесперебойная работа. 2 ГБ оперативной памяти , 16 ГБ встроенной памяти. Процессор Qualcomm Snapdragon и технология big.LITTLE – высокая производительность и низкое энергопотребление. Лёгкий запуск 3D игр и высоконагруженных программ благодаря графическому процессору",
                            en : "Fingerprint identification 2.0. Speed and accuracy: A second-generation FPC rim-free touch fingerprint sensor that offers better performance, greater accuracy, and a 0.5-second unlocking time. Adaptive sensor: Advanced technology that learns more about your fingerprint as you use it and prevents failed recognition caused by wet hands or partial fingerprints. Easy setup: Fingerprint recording in just six touches. Privacy protection: File Safe, App Lock, and Visitor Mode for added peace of mind."
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.png",
                        slug: "honor-5x"
                    },
                    "y5ii": {
                        title: "Huawei Y5 II",
                        description: {
                            hy : "?? ??? ????? ???? ????????? ????????????????: ????? ? ??????? ???????? ??? ???????? ??? ?????????????? ????????: Huawei Y5II-? ??????? ?????????? ??????? ???????? ????? ????? ???? ?????????????? ????????????: ??? 8 ?????????? ????????? F2.0 ???????????? ????????? ???????? ??????????? ????:",
                            ru : "Не упускайте возможность насладиться моментов. Жаль терять яркие моменты на фотографиях из-за освещения. Двойная вспышка Huawei Y5II поможет избежать этого в условиях низкого освещения. Камера в 8 мегапикселей и диафрагмой F2.0 придаст фотографиям профессиональный вид.",
                            en : "Never Miss a Chance to Enjoy the Moment. It’s a pity to lose vivid details in photos because of lighting. The dual flashlight of Huawei Y5II performs exceptionally well in low-light conditions. The 8 megapixel camera and the F2.0 aperture creates photos of that look professional."
                        },
                        status: 'new',
                        statusText: 'Coming Soon',
                        img: "1.jpg",
                        slug: "y5ii"
                    },
                    "huawei-y3ii": {
                        title: "Huawei Y3 II",
                        description: {
                            hy : "Easy Key ??????????: Easy Key ??????????? ???????? ? ????? ????, ????????? ??????????? ???????????????: ??????? ???????? ???????? ??????????????? ? ????? ?????????? ???",
                            ru : "ФУНКЦИЯ Easy Key. Функция Easy Key предоставит вам более удобный способ управления вашим смартфоном. Простым, двойным, либо удерживающим нажатием вы сможете открыть приложение или фукнцию, присвоенную каждому виду нажатия.",
                            en : "Music to Your Ears. Music is the irresistible drive of the night. Delivering an awesome acoustic performance, Huawei Y3II places the immersive experience of a concert in your own pocket."
                        },
                        status: ' ',
                        statusText: ' ',
                        img: "1.png",
                        slug: "huawei-y3ii"
                    },
                    "honor-5c": {
                        title: "Honor 5C",
                        description: {
                            hy : "???? ????????: ?????? ?????? 28 nm ????????????? ???????????? ??????? ???? FinFET Plus 16 nm ????????????? Kirin 650 ????????? ????????? ? ?????? ????? ??????????????????? ? ????????? ????????????????:",
                            ru : "Мощный процессор. По сравнению с процессорами предыдущего поколения с технологией 28 nm, новый чипсет Kirin 650 с технологией FinFET Plus 16 nm обеспечивает более высокую производительность и скорость работы смартфона.",
                            en : "Ultra High-Performance. Next-generation, super-powerful 16nm processor and flagship FinFET Plus technology togive you lightning speed. Leading-edge octa-core chips provide a 65% performance boost and 40% cut in power consumption."
                        },
                        status: 'new',
                        statusText: 'Coming Soon',
                        img: "1.jpg",
                        slug: "honor-5c"
                    },
                    "gr3": {
                        title: "Huawei GR3",
                        description: {
                            hy : "????????? ??? ????????? ??????: Huawei GR3 ??????????????? ? ???? ?????? ????????? ????? ???????????: ????????? ????????? ????? ????? ? ?????? ???????, ??? ??????? ??? ???? ? ????? ????????? ????:",
                            ru : "Быстрее! Лучше! С сеть LTE cat4 Huawei GR3 вы никогда не будете стоять на месте. А благодаря скорости скачивания до 150Mbps, вы будете всегда на связи с внешним миром.",
                            en : "Remarkable aluminum alloy craftsmanship. Huawei GR3 meets the modern generation’s demand for every moment to be as active as they are. An elegant frame inlaid in an aluminum alloy unibody, finished with a ceramic-sand-blasting process makes Huawei GR3 a piece of fashionable art, shining in gray, silver, and Gold."
                        },
                        status: 'new',
                        statusText: 'Coming Soon',
                        img: "1.jpg",
                        slug: "gr3"
                    },
                    "y6ii": {
                        title: "Huawei Y6 II",
                        description: {
                            hy : "??????? ?????? ????? ??????: Huawei Y6II ??? ???1280x720 ??????????????? ?????? ???????? ??? ? ?????? ??????? ???????????: GFF ????? ???????? ?????????? ???????? ?????? ??????????????? ??????? ? ??? ?????????????:",
                            ru : "Сделайте жизнь ярче! Новый динамичный дисплей Huawei Y6II с разрешением 1280x720 обеспечит вам яркость изображения на всем протежении использования. Прочная прозрачная ламинация защищает экран от царапин и иных повреждений.",
                            en : "Make Life Bolder, Brighter, More Colorful. Huawei Y6II’s new larger dynamic 1280x720 resolution display is crystal clear and rivals top digital screens, immersing you in the vivid colors and vibrancy of life in a whole new way. The durable GFF transparent lamination protects the display from scratches to ensure long-lasting brightness."
                        },
                        status: 'new',
                        statusText: 'Coming Soon',
                        img: "1.jpg",
                        slug: "y6ii"
                    },
                    "mediapad-x1-7": {
                        title: "Huawei MediaPad X1 7.0",
                        description: {
                            hy : "???????? ?????: ??? MediaPad X1 – ?? ???? ??? ?????????? ???????? ?: ??? ??????????? ? ?? ??????? ????????, ????? ?????????????????? ? ??????????? ?????????????:",
                            ru : "Ультратонкий. Новый MediaPad X1 - самый тонкий планшетный ПК в своем классе. Он расширит Ваши представления об идеальном устройстве благодаря элегантному дизайну, превосходной функциональности и удобству использования.",
                            en : "Ultra Thin. The new MediaPad X1 exceeds the limits, redefining the possibilities of design, functionality, and experience. Its ultra-slim 7.18 mm body makes it the most slender tablet in its class, while its fast LTE connectivity, advanced quad-core chip and FHD screen provide the ultimate multimedia experience, whether you’re playing, sharing, watching, or surfing."
                        },
                        img: "1.jpg",
                        slug: "mediapad-x1-7"
                    },
                    "mediapad-t1-7-0": {
                        title: "Huawei MediaPad T1 7.0",
                        description: {
                            hy : "??? ???????: IPS-?????? ???????? ? Adobe RGB ??????? ???????? ????? 90%-?, ???? ???????????? ? ????? ??? ???????? ? ???????????? ???: WSVGA ?????? ??????? (16? ?????) ???? ??????? ?? ????? ????? ??????, ??? 178° ??????????? ?????? ????????, ????? ? ????? ????? ?????? ????? ??? ????????????? ???:",
                            ru : "Яркие цвета. Красивая цветопередача. IPS-дисплей охватывает более 90% гаммы Adobe RGB, что выражается в более ярких цветах и контрастах. Благодаря WSVGA дисплею (16M цветов) вы получите более четкое изображение, а широкий угол обзора в 178° позволяет отклонять экран на большое расстояние.",
                            en : "Vibrant colors. Beautifully displayed. An oncell IPS display reproduces over 90% of Adobe RGB color space to deliver brighter colors and contrasts. With a 16M true color WSVGA screen, you can also get a clearer image display while a 178° wide-view angle makes for a naturally rotating screen."
                        },
                        img: "1.jpg",
                        slug: "mediapad-t1-7-0"
                    },
                    "mediapad-t1-8-0": {
                        title: "Huawei MediaPad T1 8.0",
                        description: {
                            hy : "????????? ??????: MediaPad T1 ???????? ???????? ?????? ? ??????????? ?????: ??????? ??????????? ??????????????? ??? ???????? ???? ??????? ????? ????? ? ????, ??? ???? ????? ?????????? ??????? ?????????, ????????? ? ????????? ?? ????????? ?????? ?????:",
                            ru : "Неповторимый дизайн. MediaPad T1, создатели которого вдохновились элегантностью лебединых крыльев, просто идеально лежит в руках. Благодаря современным технологиям, корпус планшета удалось сделать тонким и легким, а задняя часть из обработанного авиационного алюминия дополняет и без того прекрасный дизайн.",
                            en : "Tap into stunning design. Inspired by the soft and elegant shape of a Swan’s wings, the MediaPad T1 slips seamlessly into your hand. Precision technology also makes the frame slimmer and easier to use while a sandblasted, aircraft-grade aluminum back shell rounds off a perfect look."
                        },
                        img: "1.png",
                        slug: "mediapad-t1-8-0"
                    }
                };
                var prod = productS[""+req.route.slug];
                builderObject.seo = {
                    title1: "Huawei Armenia - ",
                    title2: prod.title,
                    meta_description: prod.description["" + lang],
                    // meta_keywords: keywords["" + lang],
                    url: 'http://' + req.hostname + '' + req.url,
                    img: 'https://' + req.hostname + '/img/' + prod.slug + '/' + prod.img,
                    hostname: 'http://' + req.hostname + '',
                    lang_links: [{
                        href : 'http://' + req.hostname + '/en/' + req.url.slice(4),
                        lang : 'en'
                    }, {
                        href : 'http://' + req.hostname + '/ru/' + req.url.slice(4),
                        lang : 'ru'
                    }, {
                        href : 'http://' + req.hostname + '/hy/' + req.url.slice(4),
                        lang : 'hy'
                    }],
                    social_links: [
                        'https://vk.com/huaweidevicerus',
                        'https://www.facebook.com/HuaweiDeviceRus',
                        'https://twitter.com/huaweidevicerus',
                        'http://www.youtube.com/user/HuaweiDeviceRus',
                        'http://instagram.com/huaweidevicerussia'
                    ]
                }
                path = 'products';
                break;
        }
        res.render('' + path + '.html', {data: builderObject}, function (error, html) {
            if (!error) {
                res.end(html);
                //res.end();
                return;
            }
            else {
                console.error("renderer error : " + error);
            }
            return;
        });
    }
    catch (e) {
        console.log(e);
    }
}
